thorlcr/activities/choosesets/thchoosesetsslave.cpp

```diff
@@ -249,8 +249,9 @@ class InputCounter : public CSimpleInterface, implements IThorDataLink
     virtual void getMetaInfo(ThorDataLinkMetaInfo &info);
     virtual CActivityBase *queryFromActivity();
     virtual void dataLinkSerialize(MemoryBuffer &mb);
-    unsigned __int64 queryTotalCycles() const;
-
+    virtual unsigned __int64 queryTotalCycles() const;
+    virtual void debugRequest(MemoryBuffer &msg);
+    virtual void setInputSplitterOutputBase(CSplitterOutputBase *_input, unsigned _id=0){}
```

Comment:
In NSplitterSlaveActivity, please have a look at the definition of ensureInputsConfigured(). 

Original code had this:

CDelayedInput *delayedInput = (CDelayedInput *)outputs.item(o2);

'outputs' items were IThorDataLink * but all items  in this particular 'outputs' array were created by this acitivity and it was known that these IThorDataLink * from outputs could be up cast to CDelayedInput.  This code just assumed all IThorDataLink * items in 'outputs' could be upcasted to  CDelayedInput *

When the tracing code was enabled, the assumption was no longer true.  When CDelayedInput * is added to 'outputs', if tracing is enabled, the CTracingThorDataLink (wrapping CDelayedInput) was inserted instead.  When tracing was enabled, the above code would cause delayedInput to be assigned null as it can no longer be upcast the CDelayedInput.

For each CDelayedInput it was calling setInput (i.e. delayedInput->setInput()) with a new CSplitterOutput() or new CNullInput().  

I modified the above code to 

IThorDataLink *delayedInput = (IThorDataLink *)outputs.item(o);

so that even if the 'outputs' IThorDataLink * items were CTracingThorDataLink *, it could still be assigned to delayedInput.

I added a new setInputSplitterOutputBase() virtual member function , to IThorDataLink, so that 'input' field could be assigned the new CSplitterOutput() or new CNullInput(), even if the  CTracingThorDataLink had wrapped the CDelayedInput object.  It's not ideal but I haven't come up with anything more elegant.

I renamed the original name setInput() to setInputSplitterOutputBase() as setInput was used elsewhere to set the input IThorDataLink but this was setting Owned<CSplitterOutputBase> input - a different type of object.